Source/BoundaryConditions/WarpX_PEC.H

```diff
@@ -0,0 +1,205 @@
+                                amrex::GpuArray<int, 3> const& fbndry_hi )
+    {
+        // Fields in guard cells set equal and opposite to cells in the mirror locations
+        // across the PEC boundary. Here we just initialize it.
+        amrex::IntVect ijk_mirror = ijk_vec;
+        int OnPECBoundary = 0;
+        int GuardCell = 0;
+        amrex::Real sign = 1._rt;
+#if (defined WARPX_DIM_XZ) || (defined WARPX_DIM_RZ)
+        const int sdim = ( (icomp == 1) ? 0 : 1 );
+#else
+        const int sdim = 1;
+#endif
+        // Loop over dimensions that are tangential to the Efield component, icomp.
+        // For 2D : for icomp==1, (Ey in XZ, Etheta in RZ), both X-Z and R-Z dimensions
+        //          are tangential. Therefore, we set sdim to 0 to loop over both
+        //          dimensions for this component.
+        // Otherwise : sdim = 1, to ensure we only loop over tangential dimensions.
+        for (int idim = sdim; idim < AMREX_SPACEDIM; ++idim) {
+            // dir is index of direction that is tangential to the component, icomp.
+            const int dir = ( ( (icomp+idim)<AMREX_SPACEDIM)
+                          ? (icomp+idim) : ( (icomp+idim) - AMREX_SPACEDIM));
+            // loop over sides, iside = 0 (lo), iside = 1 (hi)
+            for (int iside = 0; iside < 2; ++iside) {
+                const int is_tangent_to_PEC = ( ( iside == 0 )
+                                            ? is_boundary_PEC(fbndry_lo, dir)
+                                            : is_boundary_PEC(fbndry_hi, dir) );
+                if (is_tangent_to_PEC == 1) {
+                    // guard cell outside the domain by "ig" number cells, in direction, dir
+                    const int ig = ( (iside == 0)
+                                 ? (dom_lo[dir] - ijk_vec[dir])
+                                 : (ijk_vec[dir] - (dom_hi[dir] + stag[dir]) ) );
+
+                    if ( ig == 0) OnPECBoundary = 1;
+                    if ( ig > 0 ) {
+                        GuardCell = 1;
+                        // Mirror location inside the domain by "ig" number of cells
+                        // across PEC boundary in direction, dir, and side, iside.
+                        ijk_mirror[dir] = ( ( iside == 0)
+                                        ? (dom_lo[dir] + ig)
+                                        : (dom_hi[dir] + stag[dir] - ig));
+                        // The tangential Efield in guard cells is equal and opposite to the
+                        // value in the respective mirror location across the PEC boundary.
+                        // For corner cases, this sign change will occur for each valid
+                        // PEC boundary, and is stored in the variable "sign".
+                        sign *= -1._rt;
+                    }
+                }
+            }
+        }
+        if (GuardCell == 1) Efield(ijk_vec,n) = sign * Efield(ijk_mirror,n);
```

Comment:
Same here, use if/else if.

Reply:
I changed the order for `if(OnPECBoundary) `and `else if(GuarCell) `
because, it some cases, we could have OnPECBoundary set 1 one in direction, and Guard Cell in other.
Ie, simultaneously, both can be 1.
In this case, we always ensure that we set the value such that it satisfies the condition on the boundary.